resource/aws_codepipeline: TestAccAWSCodePipeline_deployWithServiceRole Test Failures

<!--- Please keep this note for the community --->

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

<!--- Thank you for keeping this note for the community --->

### Description

After some recent Terraform 0.12 syntax upgrades in test configurations, the following test is now consistently failing:

```
=== CONT  TestAccAWSCodePipeline_deployWithServiceRole
TestAccAWSCodePipeline_deployWithServiceRole: resource_aws_codepipeline_test.go:199: Step 1/2 error: terraform failed: exit status 1
stderr:
Error: Error creating CodePipeline: InvalidStructureException: arn:aws:iam::*******:role/codepipeline-role-v18zpiskec is not authorized to perform AssumeRole on role arn:aws:iam::*******:role/codepipeline-action-role-v18zpiskec
Error: "policy" contains an invalid JSON: invalid character 'a' looking for beginning of value
on config762278342/terraform_plugin_test.tf line 30, in resource "aws_iam_role_policy" "codepipeline_policy":
30:   policy = <<EOF
31: {
32:   "Version": "2012-10-17",
33:   "Statement": [
34:     {
35:       "Effect":"Allow",
36:       "Action": [
37:         "s3:GetObject",
38:         "s3:GetObjectVersion",
39:         "s3:GetBucketVersioning"
40:       ],
41:       "Resource": [
42:         "${aws_s3_bucket.test.arn}",
43:         "${aws_s3_bucket.test.arn}/*"
44:       ]
45:     },
46:     {
47:       "Effect": "Allow",
48:       "Action": [
49:         "codebuild:BatchGetBuilds",
50:         "codebuild:StartBuild"
51:       ],
52:       "Resource": "*"
53:     },
54:     {
55:       "Effect": "Allow",
56:       "Action": [
57:         "sts:AssumeRole"
58:       ],
59:       "Resource": aws_iam_role.codepipeline_action_role.arn
60:     }
61:   ]
62: }
63: EOF
--- FAIL: TestAccAWSCodePipeline_deployWithServiceRole (146.86s)
```

Implementing string interpolation around the `aws_iam_role.codepipeline_action_role.arn` (`"${aws_iam_role.codepipeline_action_role.arn}"`) should fix the testing and allow it pass as it did previously.

### Affected Resources

* aws_codepipeline

### Affected Tests

- TestAccAWSCodePipeline_deployWithServiceRole

### References

* [Contributing Guide on Running/Writing Acceptance Testing](https://github.com/terraform-providers/terraform-provider-aws/blob/master/docs/contributing/running-and-writing-acceptance-tests.md)

--- a/aws/resource_aws_codepipeline_test.go
+++ b/aws/resource_aws_codepipeline_test.go
@@ -685,7 +685,7 @@ resource "aws_iam_role_policy" "codepipeline_policy" {
       "Action": [
         "sts:AssumeRole"
       ],
-      "Resource": aws_iam_role.codepipeline_action_role.arn
+      "Resource": "${aws_iam_role.codepipeline_action_role.arn}"
     }
   ]
 }

